Mage/src/main/java/mage/abilities/effects/common/CreateTokenEffect.java

```diff
@@ -55,11 +57,28 @@ public CreateTokenEffect(Token token, int amount, boolean tapped, boolean attack
     }
 
     public CreateTokenEffect(Token token, DynamicValue amount, boolean tapped, boolean attacking) {
+        this(token, amount, tapped, attacking, (List<Token>)null);
+    }
+
+    public CreateTokenEffect(Token token, int amount, boolean tapped, boolean attacking, Token additionalToken) {
+        this(token, StaticValue.get(amount), tapped, attacking, Arrays.asList(additionalToken));
+    }
+
+    public CreateTokenEffect(Token token, DynamicValue amount, boolean tapped, boolean attacking, Token additionalToken) {
+        this(token, amount, tapped, attacking, Arrays.asList(additionalToken));
+    }
+
+    public CreateTokenEffect(Token token, int amount, boolean tapped, boolean attacking, List<Token> additionalTokens) {
+        this(token, StaticValue.get(amount), tapped, attacking, additionalTokens);
+    }
+
+    public CreateTokenEffect(Token token, DynamicValue amount, boolean tapped, boolean attacking, List<Token> additionalTokens) {
```

Comment:
Rather than adding all this clutter to the constructors, how about the following:
- The field that stores additional tokens is initialized to a `final` `new ArrayList()`
- A chaining method `public CreateTokenEffect withAdditionalTokens(Token... tokens)` that adds those tokens (one or more) to the internal list
- Check `isEmpty()` rather than null for all usage of the list of additional tokens

This protects the internal data structure, avoids null usage, and should be cleaner when writing card code.

Reply:
Good call, that's a lot cleaner.